src/http/ureq.rs

```diff
@@ -14,35 +14,44 @@ impl ClientError {
 }
 
 impl Spotify {
+    /// The request handling in ureq is split in three parts:
+    ///
+    /// * The initial request (POST, GET, ...) is given as the `request`
+    ///   parameter.
+    /// * This method will add whichever headers and additional data is needed
+    ///   for all requests.
+    /// * The request is finished and performed with the `send_request` function
+    ///   (JSON, a form...).
     fn request<D>(
         &self,
-        request: &mut Request,
+        mut request: Request,
```

Comment:
Why would you like to change the `request` parameter from `&mut` to `mut`?

Reply:
ureq's builder pattern no longer uses `&mut Request`, it's now `Request`. So instead of using `&mut Request` and calling the builder with `builder.method()`, it's now `builder = builder.method()`, which requires `request` to be mutable in the function.